src/js/components/ServicePlanBlockDetails.js

```diff
@@ -0,0 +1,204 @@
+      if (isComplete && hasDecisionPoint) {
+        hasDecisionPoint = false;
+      }
+
+      let blockClassName = classNames('service-plan-modal-details-block', {
+        'has-decision-point': hasDecisionPoint,
+        'is-active': isActive,
+        'is-selected': selectedBlock && selectedBlock.getID() === block.getID(),
+        'is-complete': isComplete
+      });
+
+      let blockIcon = <IconUpgradeBlock />
+
+      if (hasDecisionPoint) {
+        blockIcon = <IconUpgradeBlockDecisionPoint />;
+      }
+
+      return (
+        <div className={blockClassName} key={blockIndex}
+          onClick={this.handleBlockClick.bind(this, block)}
+          onMouseEnter={this.handleBlockMouseEnter.bind(this, block)}
+          onMouseLeave={this.handleBlockMouseLeave}>
+          <Tooltip
+            content={this.getUpgradeBlockTooltipContent({
+              block,
+              blockIndex,
+              activePhase
+            })}
+            interactive={true}
+            ref={block.getID()}
+            stayOpen={true}
+            suppress={true}>
+            <div className="service-plan-modal-details-block-content">
+              {blockIcon}
+            </div>
+          </Tooltip>
+        </div>
+      );
+    });
+
+    return (
+      <div className="service-plan-modal-details-blocks">
+        {blocks}
+      </div>
+    );
+  }
+
+  getUpgradeBlockTooltipContent(blockDetails) {
+    let {activePhase, block, blockIndex} = blockDetails;
+    let activePhaseID = activePhase.getID();
+    let blockID = block.getID();
+    let blockOptions = [];
+    let blockLabelSeparator = ': ';
+
+    if (block.isComplete() || block.isInProgress()) {
+      blockOptions.push(
+        <a
+          className="clickable"
+          key="restart"
+          onClick={this.handleBlockRestart.bind(
+            this,
+            blockID,
+            activePhaseID
+          )}>
+          Restart
+        </a>
+      );
+    }
+
+    if (!block.isComplete() || block.isPending()) {
+      blockOptions.push(
+        <a
+          className="clickable"
+          key="force-complete"
+          onClick={this.handleBlockForceComplete.bind(
+            this,
+            blockID,
+            activePhaseID
+          )}>
+          Force Complete
+        </a>
+      );
+    }
+
+    return (
+      <span className="service-plan-modal-details-block-tooltip">
+        <span className="service-plan-modal-details-block-label">
+          <strong>Block {blockIndex + 1}</strong>{blockLabelSeparator}
+        </span>
+        {blockOptions}
+      </span>
+    );
+  }
+
+  render() {
+    let {servicePlan} = this.props;
+    let activePhase = servicePlan.getPhases().getActive();
+    let phaseBlocks = activePhase.getBlocks();
+    let blockCount = phaseBlocks.getItems().length;
+
+    let blocksHeadingContent = `Completed ${phaseBlocks.getComplete().length}
+      of ${blockCount}`;
+
+    if (!!this.state.hoveredBlock || !!this.state.selectedBlock) {
```

Comment:
let's store these state properties into variables. you can simple do `let {hoveredBlock, selectedBlock} = this.state` on top